packages/web-components/src/dropdown/dropdown.ts

```diff
@@ -0,0 +1,1044 @@
+    this.indicatorSlot?.assign(this.indicator);
+  }
+
+  /**
+   * Handles the connected event for the listbox.
+   *
+   * @param e - the event object
+   * @internal
+   */
+  private listboxConnectedHandler(e: Event): void {
+    const target = e.target as HTMLElement;
+
+    if (isListbox(target)) {
+      this.listbox = target;
+    }
+  }
+
+  /**
+   * Adds or removes the window event listener based on the open state.
+   *
+   * @param prev - the previous open state
+   * @param next - the current open state
+   * @internal
+   */
+  public openChanged(prev: boolean | undefined, next: boolean | undefined): void {
+    super.openChanged(prev, next);
+
+    if (next) {
+      Dropdown.AnchorPositionFallbackObserver?.observe(this.listbox);
+      window.addEventListener('scroll', this.scrollWindowHandler, { passive: true });
+      this.scrollWindowHandler();
+      return;
+    }
+
+    Dropdown.AnchorPositionFallbackObserver?.unobserve(this.listbox);
+    window.removeEventListener('scroll', this.scrollWindowHandler);
+  }
+
+  /**
+   * Applies anchor positioning fallback styles.
+   *
+   * @internal
+   */
+  private setAnchorPositionFallbackStyles(): void {
+    const anchorName = `--${this.id}`;
+
+    if (AnchorPositioningCSSSupported) {
+      return;
+    }
+
+    const observerCallback = (entries: IntersectionObserverEntry[]): void => {
+      entries.forEach(entry => {
+        const target = entry.target as Listbox;
+        if (isListbox(target)) {
+          if (inRange(entry.intersectionRatio, 0, 1)) {
+            toggleState(
+              target.dropdown?.elementInternals,
+              'flip-block',
+              entry.intersectionRect.bottom >= window.innerHeight,
+            );
+          }
+        }
+      });
+    };
+
+    Dropdown.AnchorPositionFallbackObserver =
+      Dropdown.AnchorPositionFallbackObserver ?? new IntersectionObserver(observerCallback, { threshold: [0, 1] });
+
+    toggleState(this.elementInternals, 'anchor-position-fallback', true);
+
+    if (!window.CSS_ANCHOR_POLYFILL) {
+      return;
+    }
+
+    if (Dropdown.AnchorPositionFallbackTimeout) {
+      cancelIdleCallback(Dropdown.AnchorPositionFallbackTimeout);
+      Dropdown.AnchorPositionFallbackTimeout = null;
+    }
+
+    this.anchorPositionFallbackStyleElement =
+      this.anchorPositionFallbackStyleElement ?? document.createElement('style');
+
+    this.anchorPositionFallbackStyleElement.textContent = /* css */ `
+      #${this.id} { anchor-name: ${anchorName}; }
+
+      #${this.listbox.id} {
+        position-anchor: ${anchorName};
+        position-try-fallbacks: block-start, flip-inline, flip-block;
+        max-height: var(--listbox-max-height, calc(100vh - anchor-size(self-block)));
+        min-width: anchor-size(width);
+      }
+    `;
+
+    Dropdown.AnchorPositionFallbackStyleElements.set(this, this.anchorPositionFallbackStyleElement);
+
+    Dropdown.AnchorPositionFallbackTimeout = requestIdleCallback(
```

Comment:
Safari doesn't have `requestIdleCallback` yet 😭 only in Technical Preview as of now. Maybe just use `requestAnimationFrame` or `setTimeout`?

Reply:
It's ponyfilled in [`packages/web-components/src/utils/idle-callback.ts`](https://github.com/microsoft/fluentui/pull/32116/files/3761ed8e608511c77ba0a08a1c858d97f060fcc0#diff-516be0235c645c65481415b0cbf0020f442e15657d1912ee1e50d5874578cf98)

Reply:
I removed the routine that used this, so this has been removed as well.